src/theory/sets/skolem_cache.cpp

```diff
@@ -35,8 +35,19 @@ Node SkolemCache::mkTypedSkolemCached(
   std::map<SkolemId, Node>::iterator it = d_skolemCache[a][b].find(id);
   if (it == d_skolemCache[a][b].end())
   {
-    Node sk = mkTypedSkolem(tn, c);
+    SkolemManager* sm = NodeManager::currentNM()->getSkolemManager();
+    Node sk;
+    if (id == SkolemId::SK_PURIFY)
+    {
+      Assert(a.getType() == tn);
+      sk = sm->mkPurifySkolem(a, c);
+    }
+    else
+    {
+      sk = sm->mkDummySkolem(c, tn, "sets skolem");
+    }
     d_skolemCache[a][b][id] = sk;
+    d_allSkolems.insert(sk);
```

Comment:
Is this needed? It seems this was working before.  The function `isSkolem` which uses this private field `d_allSkolems` is not used anywhere.

Reply:
isSkolem is used in sets solver state still.

I initially thought this wasnt needed but this does impact how the skolems are handled.